### PR TITLE
VZ-5030: Update readme to explain upgrade proceedure

### PR DESCRIPTION
**Pull request by mbenmclean**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/master/CONTRIBUTING.md) guide
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by gilbode — COMMENTED** [review]
Move this to BUILD_FROM_SOURCE_README - that's the one we will read that wont' get inter-mixed w/ content from upstream.

**Comment by mbenmclean** [comment]
Sounds good

**Review by wmhopkins — COMMENTED** [review]
Might be worth explicitly mentioning that this commit is the ingress template file, and that it may need to be modified to reflect changes made to the underlying template file in nginx.

**Comment by mbenmclean** [comment]
Ok sounds good. The note might get lost after the upgrade but can't hurt

File: BUILD_FROM_SOURCE_README.md [hunk 1]
```diff
@@ -15,3 +15,9 @@ export DOCKER_TAG=<Image Tag>
 ```
 ./build-images.sh
 ```
+
+# Notes
+When upgrading NGINX Ingress Controller, an additional step must be taken. After creating a new branch for the newer version, you
```
Comment (rosemarymarano) [inline BUILD_FROM_SOURCE_README.md:20]: an additional step must be taken -> you must take an additional step.
**I find your use of new and newer confusing, so here are my suggestions to provide clarity** ->
After creating a new branch for the newer version -> After creating a new branch for the version upgrade OR After creating a new branch for the upgraded version (use whichever one you think is more clear)

File: BUILD_FROM_SOURCE_README.md [hunk 2]
```diff
@@ -15,3 +15,9 @@ export DOCKER_TAG=<Image Tag>
...
+must cherrypick commit [ebe940c](https://github.com/verrazzano/ingress-nginx/commit/ebe940c0c38f8137b5bbbd5bf2689cec0bfff06b)
+to the newer branch. This commit made a code change to the nginx.tmpl file and must be included in later releases in order for
```
Comment (rosemarymarano) [inline BUILD_FROM_SOURCE_README.md:22]: to the newer branch. -> to the new branch. 
This commit made -> This commit makes
the nginx.tmpl file -> the `nginx.tmpl` (code font) file

File: BUILD_FROM_SOURCE_README.md [hunk 3]
```diff
@@ -15,3 +15,9 @@ export DOCKER_TAG=<Image Tag>
...
+nginx error pages to work as expected.
```
Comment (rosemarymarano) [inline BUILD_FROM_SOURCE_README.md:23]: in order for nginx error pages -> for NGINX error pages

**Review by rosemarymarano — COMMENTED** [review]
Some edits.

File: BUILD_FROM_SOURCE_README.md (hunk 1 above)
Comment (mbenmclean) [reply BUILD_FROM_SOURCE_README.md:20]: How about After creating a branch for the new NGINX version

**Pull request merged**